internal/plugins/nfs-shares.go

```diff
@@ -23,6 +25,45 @@ import (
 	"github.com/sapcc/castellum/internal/db"
 )
 
+var (
+	nfsTypeRx         = regexp.MustCompile(`^[A-Za-z0-9-]+$`)
+	shareTypeIDToName map[string]string
+)
+
+type assetTypeNFS struct {
+	AllShares bool
+	NFSType   string
+}
+
+func (m *assetManagerNFS) parseAssetType(assetType db.AssetType) *assetTypeNFS {
+	if assetType == "nfs-shares" {
+		return &assetTypeNFS{AllShares: true}
+	}
+	if strings.HasPrefix(string(assetType), "nfs-shares-type:") {
+		nfsType := strings.TrimPrefix(string(assetType), "nfs-shares-type:")
+		if nfsTypeRx.MatchString(nfsType) {
```

Comment:
We should only accept share type names that actually exist, i.e. this should check `shareTypeIDToName` (and then the regexp is not needed).